src/binary_tree.rs

```diff
@@ -0,0 +1,380 @@
+            Ordering::Greater => match &mut self.right {
+                Some(node) => node.insert(value),
+                None => {
+                    self.right = Some(Box::new(Node::new(value)));
+                    return true;
+                }
+            },
+        }
+    }
+
+    fn find(&self, value: T) -> bool {
+        match value.cmp(&self.value) {
+            Ordering::Equal => true,
+            Ordering::Greater => match self.right.as_ref() {
+                Some(node) => node.find(value),
+                None => false,
+            },
+            Ordering::Less => match self.left.as_ref() {
+                Some(node) => node.find(value),
+                None => false,
+            },
+        }
+    }
+
+    pub fn inorder(&self) -> Vec<&T> {
+        let mut result = vec![];
+        match self.left.as_ref() {
+            Some(node) => {
+                let mut left_vec = node.inorder();
+                result.append(left_vec.as_mut());
+            }
+            None => (),
+        }
+        result.push(&self.value);
+        match self.right.as_ref() {
+            Some(node) => {
+                let mut right_vec = node.inorder();
+                result.append(right_vec.as_mut());
+            }
+            None => (),
+        }
+        result
+    }
+
+    pub fn postorder(&self) -> Vec<&T> {
+        let mut result = vec![];
+        match self.left.as_ref() {
+            Some(node) => {
+                let mut vec_left = node.postorder();
+                result.append(vec_left.as_mut());
+            }
+            None => (),
+        }
+        match self.right.as_ref() {
+            Some(node) => {
+                let mut right_vec = node.postorder();
+                result.append(right_vec.as_mut());
+            }
+            None => (),
+        }
+        result.push(&self.value);
+        result
+    }
+
+    pub fn preorder(&self) -> Vec<&T> {
+        let mut result = vec![];
+        result.push(&self.value);
+        match self.left.as_ref() {
+            Some(node) => {
+                let mut vec_left = node.preorder();
+                result.append(vec_left.as_mut());
+            }
+            None => (),
+        }
+        match self.right.as_ref() {
+            Some(node) => {
+                let mut right_vec = node.preorder();
+                result.append(right_vec.as_mut());
+            }
+            None => (),
+        }
+        result
+    }
+}
+
+impl<'a, T: std::cmp::Eq> Iterator for Iter<'a, T> {
+    type Item = &'a T;
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.visited.is_empty() {
+            return None;
+        }
+
+        let node = self.visited.pop();
+        match (
+            node.unwrap().as_ref().left.as_ref(),
+            node.unwrap().as_ref().right.as_ref(),
+        ) {
+            (None, None) => (),
+            (leaf @ Some(_), None) | (None, leaf @ Some(_)) => self.visited.push(&leaf.unwrap()),
```

Comment:
Magic!!

Reply:
You can resolve this, comment i just wanted you to see it 🥇

Reply:
😂😂😂